agent/consul/leader_connect_ca.go

```diff
@@ -510,6 +512,7 @@ func (c *CAManager) initializeRootCA(provider ca.Provider, conf *structs.CAConfi
 // to non-ready.
 func (c *CAManager) initializeSecondaryCA(provider ca.Provider, config *structs.CAConfiguration) error {
 	activeIntermediate, err := provider.ActiveIntermediate()
+	activeIntermediate = ca.AddSingleNewline(activeIntermediate)
```

Comment:
Is this necessary? We seem to only be parsing it here, not storing it.

If we are storing this, maybe we can move this line to immediately before we set it on the struct, so it's clear what it's for?